airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/base.py

```diff
@@ -119,6 +122,11 @@ def should_retry(self, response: requests.Response) -> bool:
         if response.status_code == 402:
             self.logger.warning(f"Unable to perform a request. Payment Required: {response.json()['error']}")
             return False
+        if response.status_code == 400 and "Unable to authenticate request" in response.text:
+            self.logger.warning("Your credentials might have expired. Please update your config with valid credentials.")
+            setattr(self, "raise_on_http_errors", False)
+            self._skip_stream = True
```

Comment:
why it can't be a part of HttpAvailabilityStrategy, it's not suitable here? we can extend `reasons_for_unavailable_status_codes` method to add 400 status code, so you don't need a  `_skip_stream`.

Reply:
The reason the availability strategy is not suitable here is that we can face this error any time, not just when reading the first record of the first slice. On the other hand, now I'm not sure if we need this at all because such an approach allows invalid config to pass the connectivity check..

Reply:
reverted